app/utils.py

```diff
@@ -0,0 +1,27 @@
+from pydantic import BaseModel
+
+
+def union(source, destination):
+    for key, value in source.items():
+        if isinstance(value, dict):
+            node = destination.setdefault(key, {})
+            union(value, node)
+        else:
+            destination[key] = value
+
+    return destination
+
+
+def remove_optional_unset_args(model: BaseModel):
```

Comment:
add that stackovrflow link as a comment please

and ye union waala function ko andar hi likh de na, inside `remove_optional_unset_args`?
For now that is going to be used here only.

If we need it kabhi we can take it out.
It will just be easier to understand for someone who sees the code after an year